plugins/telescope/extensions/undo.nix

```diff
@@ -77,9 +94,11 @@ telescopeHelpers.mkExtension {
       the side-by-side view even if this is set to false.
     '';
 
-    diff_context_lines = helpers.defaultNullOpts.mkStrLuaOr types.ints.unsigned "vim.o.scrolloff" ''
-      Defaults to the scrolloff.
-    '';
+    vim_diff_opts = {
```

Comment:
Ideally, this would be a `attrsOf anything`, but it would make the typing of `ctxlen` as well as the rename more annoying.

Reply:
Can we use a freeformType to achieve that?

I was under the impression that using nested attrs like this in option declaration was rfc42 compliant?

Or is this not the case here because it isn't a `settings` option (and hence not already in a freeform module)?

Reply:
I guess it's fine like this.
The `freeformType` from the parent `settings` option lets a user define any `settings.vim_diff_opts.*` so functionally it's ok.

Reply:
Ah I see. Should we have a warning when cfg.settings has the attr `diff_context_lines` defined?

Reply:
Hmm I don't think so, because the plugin is doing that already.

Reply:
Up to you. If there's a runtime warning we probably don't need to worry to much.

It's still nice to have a warning when building, but that's additional maintenance cost.